src/main/java/com/google/sps/data/ComputeAPIClientInterface.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.data;
+
+import com.google.api.services.compute.model.Operation;
+import java.io.IOException;
+import java.util.Optional;
+
+/** Interface for the ComputerAPIClient class. */
+public interface ComputeAPIClientInterface {
+  /** 
+   * Creates a new Instance.
+   * @param {String} instanceName - the name of the new Instance.
+   * @param {String} vncServerPassword - the password needed to access 
+   *    to vnc server.
+   */
+  void createInstance(String instanceName, String vncServerPassword) 
```

Comment:
Something to think about, but not necessarily required. Adding @chukwuemekagooglecorp for thoughts.

All of your implementations perform the operation then immediately wait. A user of your client may not necessarily want to do this. You could make waitForOperationComplete(Operation op) a public method which takes an operation as input, and have each of these functions return the operation they execute. A user can then decide if they want to wait immediately or not.

Theres a lot of ways to design this and we probably don't have time to implement it all, so this is just food for thought. I think for a minimum product, automatically blocking is fine.

Reply:
Yeah, if we had time, i'd elect we add the Operation as a return value and include a parameter "boolean waitUntilComplete". So something like:
Operation createInstance(..., boolean waitUntilComplete); 
If waitUntilComplete is true, it automatically waits, otherwise it doesn't. 

But, like Danny said, what you have is fine.